tools/publish-lib.js

```diff
@@ -71,17 +71,20 @@ async function main() {
   await writeLib('main', `${lib}-nightly`);
   console.log('  done');
 
-  console.log('Update README and LICENSE if needed...');
-  for (const file of ['README.md', 'LICENSE']) {
-    const { stdout: fileOut } = await execFile('git', ['show', `${lastTag}:${file}`]);
-    fs.writeFileSync(file, fileOut, 'utf8');
+  console.log('Update README, LICENSE and demo.html if needed...');
+  for (const file of ['README.md', 'LICENSE', {inpath: 'test/resources/demo.html', outpath: 'demo.html'}]) {
+    const inpath = file.inpath ?? file;
+    const outpath = file.outpath ?? file;
+    const { stdout: fileOut } = await execFile('git', ['show', `${lastTag}:${inpath}`]);
```

Comment:
using `main` instead of `${lastTag}` should solve the problem and still achieve what we want, right?

Reply:
That depends on "what we want" :)

My thinking was that we would want to publish the latest **released** versions of these files, and not the nightly versions.

That said, to fix our immediate issue, we don't really need to make any change to the code. We can just as well publish a new tag, which will become `${lastTag}`.

Reply:
> My thinking was that we would want to publish the latest **released** versions of these files, and not the nightly versions.

I guess my thinking is that for the license, the README or the demo, there isn't necessarily much value in tying our hands that way?

In other words, it feels to me more likely that we will want to update the README info or the demo (to make them more complete, good-looking, ...) between releases than it is likely that these updates would be tightly coupled with releases.

Reply:
> I guess my thinking is that for the license, the README or the demo, there isn't necessarily much value in tying our hands that way?

To some extent. I agree that's optimistic but, ideally, when a PR adds a feature, it will also update the README and the demo at the same time to document and showcase the feature, and we would want to publish the updates altogether, not to document a feature that has not yet been released.

Reply:
That being said, the new feature will exist in the nightly version...
Versioning management is a pain :)

Reply:
both good points :)

Very practically speaking, the demo file import `https://w3c.github.io/i-slide/i-slide-1.js` which couples it strongly to 1.x, so is in contradiction with my earlier claim :)

Let's go with tying it with releases for the time being, and we can revisit this later if this proves too constraining.